modules/git/blame.go

```diff
@@ -131,15 +152,44 @@ func CreateBlameReader(ctx context.Context, repoPath, commitID, file string) (*B
 		if err != nil {
 			log.Error("Error running git blame (dir: %v): %v, stderr: %v", repoPath, err, stderr.String())
 		}
-	}(cmd, repoPath, stdout, done)
+	}()
 
 	bufferedReader := bufio.NewReader(reader)
 
 	return &BlameReader{
-		cmd:            cmd,
 		output:         stdout,
 		reader:         reader,
 		bufferedReader: bufferedReader,
 		done:           done,
+		ignoreRevsFile: ignoreRevsFile,
 	}, nil
 }
+
+func tryCreateBlameIgnoreRevsFile(commit *Commit) *os.File {
+	entry, err := commit.GetTreeEntryByPath(".git-blame-ignore-revs")
+	if err != nil {
+		return nil
+	}
+
+	r, err := entry.Blob().DataAsync()
+	if err != nil {
+		return nil
+	}
+	defer r.Close()
+
+	f, err := os.CreateTemp("", "gitea_git-blame-ignore-revs")
+	if err != nil {
+		return nil
+	}
+	defer f.Close()
+
+	_, err = io.Copy(f, r)
+	if err != nil {
+		defer func() {
+			_ = util.Remove(f.Name())
+		}()
+		return nil
+	}
```

Comment:
Without "defer", it seems easier to read (just a nit).

```suggestion
	f, err := os.CreateTemp("", "gitea_git-blame-ignore-revs")
	if err != nil {
		return nil
	}
	_, err = io.Copy(f, r)
	_ = f.Close()
	if err != nil {
		_ = util.Remove(f.Name())
		return nil
	}
```